test/integration/controller/core/queuedworkload_controller_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"fmt"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kueue "sigs.k8s.io/kueue/api/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/integration/framework"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("QueuedWorkload controller", func() {
+	var (
+		ns                   *corev1.Namespace
+		updatedQueueWorkload kueue.QueuedWorkload
+		queue                *kueue.Queue
+		wl                   *kueue.QueuedWorkload
+		message              string
+	)
+
+	ginkgo.When("the queue is not defined in the workload", func() {
+		ginkgo.BeforeEach(func() {
```

Comment:
this `BeforeEach` could be outside of `When`, since all the cases need to create a namespace.